app/components/Form/ProjectEmissionIntensityReportForm.tsx

```diff
@@ -203,14 +216,39 @@ const ProjectEmissionsIntensityReport: React.FC<Props> = (props) => {
     isStagingEmissionIntensity,
   ] = useStageEmissionIntensityFormChange();
 
+  const formData = emissionIntensityReportFormChange?.newFormData;
+
+  // Restructure form data to be in the same format as the schema
+  const formattedFormData = {
+    teimpReporting: {},
+    uponCompletion: {},
+  };
+
+  Object.keys(emissionIntensityReportSchema.properties).forEach(
+    (schemaProperty) =>
+      Object.keys(
+        emissionIntensityReportSchema.properties[schemaProperty].properties
+      ).forEach(
+        (key) =>
+          formData?.[key] &&
+          Object.assign(formattedFormData[schemaProperty], {
+            [key]: formData[key],
+          })
+      )
+  );
```

Comment:
We can likely refactor this to a much simpler front-end code, and offload the flattening / parsing of the data to a custom commit handler on the database instead. I'd put a comment here to that effect, and maybe a tech debt card with low priority

Let me know if you need help writing these!

Reply:
Captured in https://github.com/bcgov/cas-cif/issues/1373